File: 06_Algorithm/leetcode_429.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_429.py
# Author            : Andy
# Date              : 2022.08.21
# Last Modified Date: 2022.08.21
# Last Modified By  : Andy


class Solution:
    def levelOrder(self, root):
        result = []
        q = []
        if not root: return result

        q.append(root)

        while q:
            size = len(q)
            cur = []
            for i in range(size):
                node = q[0]
                del q[0]
                cur.append(node.val)
                for child in node.children:
                    q.append(child)
            result.append(cur)
        return result

File: 06_Algorithm/leetcode_429.go
/**
 * File              : leetcode_429.go
 * Author            : Andy
 * Date              : 2022.08.21
 * Last Modified Date: 2022.08.21
 * Last Modified By  : Andy
 */

package levelOrder

func levelOrder(root *Node) [][]int {
	var result [][]int
	q := make([]*Node, 0)
	if root == nil {
		return result
	}
	q = append(q, root)
	for len(q) != 0 {
		// 用来存当前层的数据
		cur := make([]int, 0)
		size := len(q)
		for i := 0; i < size; i++ {
			// 从前面取出一个节点
			node := q[0]
			q = q[i:]
			cur = append(cur, node.Val)
			for _, child := range node.Children {
				q = append(q, child)
			}
		}
		result = append(result, cur)
	}
	return result
}
